src/main/fc/config.c

```diff
@@ -268,27 +269,45 @@ void resetServoConfig(servoConfig_t *servoConfig)
 
 void resetMotorConfig(motorConfig_t *motorConfig)
 {
+    int motorIndex = 0;
+    for (int i = 0; i < USABLE_TIMER_CHANNEL_COUNT && motorIndex < MAX_SUPPORTED_MOTORS; i++) {
+        if (timerHardware[i].usageFlags & TIM_USE_MOTOR) {
+            motorConfig->ioTags[motorIndex] = timerHardware[i].tag;
+            motorIndex++;
+        }
+    }
+
 #ifdef BRUSHED_MOTORS
     motorConfig->minthrottle = 1000;
     motorConfig->motorPwmRate = BRUSHED_MOTORS_PWM_RATE;
     motorConfig->motorPwmProtocol = PWM_TYPE_BRUSHED;
     motorConfig->useUnsyncedPwm = true;
 #else
-    motorConfig->minthrottle = 1070;
-    motorConfig->motorPwmRate = BRUSHLESS_MOTORS_PWM_RATE;
-    motorConfig->motorPwmProtocol = PWM_TYPE_ONESHOT125;
+#ifdef BRUSHED_ESC_AUTODETECT
+    IO_t MotorDetectPin = IOGetByTag(motorConfig->ioTags[0]);
+    IOInit(MotorDetectPin, OWNER_SYSTEM, 0);
+    IOConfigGPIO(MotorDetectPin, IOCFG_IPU);
```

Comment:
Is it safe to do this at runtime, after initialisation? `resetMotorConfig` can be called as part of CLI `diff`, or when resetting the config to defaults. For that reason I liked the two part implementation for the ALIENFLIGHT boards better, where the hardware detection was done only once during board initialisation.